src/building/construction.c

```diff
@@ -662,9 +663,7 @@ void building_construction_cancel(void)
 {
     map_property_clear_constructing_and_deleted();
     if (data.in_progress && building_construction_is_updatable()) {
-        if (building_construction_is_updatable()) {
-            game_undo_restore_map(1);
-        }
+        game_undo_restore_map(1);
```

Comment:
Why this change?

Reply:
building_construction_is_updatable() has to be true in order to make it into this if block, so checking here again is redundant.